modules/private.php

```diff
@@ -0,0 +1,256 @@
+			return;
+		}
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			return;
+		}
+
+		// Serve robots.txt for private blogs.
+		if ( is_object( $wp ) && ! empty( $wp->query_vars['robots'] ) ) {
+			return;
+		}
+
+		if ( $current_user && ( is_super_admin() || Jetpack_Private::is_private_blog_user( $wpdb->blogid, $current_user ) ) ) {
+			return;
+		}
+
+		if ( file_exists( JETPACK__PLUGIN_DIR . '/modules/private/private.php' ) ) {
+			include JETPACK__PLUGIN_DIR . '/modules/private/private.php';
+		} else {
+			esc_html_e( 'This site is private.', 'jetpack' );
+		}
+
+		exit;
+	}
+
+	/**
+	 * Does not check whether the blog is private. Accepts blog and user in various types.
+	 * Returns true for super admins; if you don't want that, use is_really_private_blog_user.
+	 *
+	 * @param int   $blog Current WordPress blod id.
+	 * @param Mixed $user Current WordPress user or user_id.
+	 */
+	static function is_private_blog_user( $blog, $user ) {
+		global $wpdb;
+
+		if ( ! is_object( $user ) ) {
+			$user = new WP_User( $user );
+		}
+
+		if ( ! $user->ID ) {
+			return false;
+		}
+
+		if ( is_numeric( $blog ) ) {
+			$blog_id = intval( $blog );
+		} elseif ( is_object( $blog ) ) {
+			$blog_id = $blog->blog_id;
+		} elseif ( is_string( $blog ) ) {
+			$blog    = get_blog_info( $blog, '/', 1 );
+			$blog_id = $blog->blog_id;
+		} else {
+			$blog_id = $wpdb->blogid;
+		}
+
+		// check if the user has read permissions.
+		$the_user = wp_clone( $user );
+		$the_user->for_site( $blog_id );
+		return $the_user->has_cap( 'read' );
+	}
+
+	/**
+	 * Hides the blog's name on the login form for private blogs.
+	 */
+	static function privatize_blog_maybe_mask_blog_name() {
+		add_filter( 'bloginfo', array( __CLASS__, 'privatize_blog_mask_blog_name' ), 3, 2 );
+	}
+
+	/**
+	 * Replaces the the blog's "name" value with "Protected Blog"
+	 *
+	 * @see privatize_blog_maybe_mask_blog_name()
+	 * @param mixed $value The requested non-URL site information.
+	 * @param mixed $what  Type of information requested.
+	 */
+	static function privatize_blog_mask_blog_name( $value, $what ) {
+		if ( in_array( $what, array( 'name', 'title' ), true ) ) {
+			$value = __( 'Protected Blog', 'jetpack' );
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Filters new comments so that users can't comment on private blogs
+	 *
+	 * @param array $comment Documented in wp-includes/comment.php.
+	 */
+	static function privatize_blog_comments( $comment ) {
+		Jetpack_Private::privatize_blog( null );
+		return $comment;
+	}
+
+	/**
+	 * Extend the 'Site Visibility' privacy options to also include a private option
+	 **/
+	static function privatize_blog_priv_selector() {
+		?>
+		<style>
+			.jetpack-private__setting-disabled {
+				font-weight: bold;
+				margin-top: 10px;
+				padding: 10px;
+			}
+		</style>
+		<div class="jetpack-private__setting-disabled highlight">
+        	<?php _e( 'This setting is ignored because you made your site private in Jetpack', 'jetpack' ); ?>
```

Comment:
Should we add a link to the settings here, using the search term so we get the setting right away? This way we could replace "in Jetpack" by something more explicit and more informative. `wp-admin/admin.php?page=jetpack#/settings?term=private`

Reply:
Good idea, done 👍